webrender/src/resource_cache.rs

```diff
@@ -254,6 +256,16 @@ impl ResourceCache {
         self.texture_cache.max_texture_size()
     }
 
+    fn should_tile(&self, descriptor: &ImageDescriptor, data: &ImageData) -> bool {
+        let limit = self.max_texture_size();
+        return match data {
+            // Tiled external images are not implemented.
+            &ImageData::ExternalHandle(_) => false,
+            &ImageData::ExternalBuffer(_) => false,
+            _ => { descriptor.width > limit || descriptor.height > limit }
```

Comment:
As a follow up, could we explicitly list all the variations here rather than _ ? I like the safety of a compiler bug when new enum variants are introduced.